pegasus/sites.v3/code.org/views/top_hoc_tutorials.haml

```diff
@@ -1,6 +1,6 @@
 = view :top_hoc_tutorial_applab, image_match_youtube_height: false
 
-.col-50
+.col-49
```

Comment:
Is there any way to avoid a special 49% width?  Sometimes `box-sizing: border-box` can bring things back into bounds, but sometimes I guess it's necessary...

Reply:
i like using 49% for flex elements that take up 50% but need a little padding, so i can see us using this in the future.

alternatively, both elements could use `.col-50`, then add special padding rules for the left and right elements so that the outer alignment is unaffected

Reply:
I suspect this might be related?  https://github.com/w3c/csswg-drafts/issues/592

I've been repeatedly disappointed by this in Flexbox, and am wondering if CSS Grid might be a better avenue to pursue long term.

The problem with something like 49% for a two column layout is that you need exactly 2% for the gap if you want both left and right edges to align, and it requires similarly careful math once the UI is responsive, along with new locations of gaps to maintain alignment on both left and right edges.  Flexbox should have solved it completely, but the lack of `flex-gap` has left it wanting.

Reply:
i didn't know flexbox adopted row- and column-gap! i'll mess around with that and make a follow-up PR.

i _really_ like CSS grid for static content, but found it a little less flexible for dynamic content (i think most likely due to my relative inexperience with it versus flexbox, though). i would be in favor of using CSS grid moving forward -- do you know if this is something we're trying to move to?

Reply:
Unfortunately it looks like support might not be universal?  https://developer.mozilla.org/en-US/docs/Web/CSS/column-gap#Browser_compatibility

I've not used CSS grid myself, but it'd be worth an investigation.  I just know that Flex got us closer to success in a number of cases, but hadn't realised that `column-gap` was actually implemented widely enough to maybe get us all the way.  If it is, then that would be good news.